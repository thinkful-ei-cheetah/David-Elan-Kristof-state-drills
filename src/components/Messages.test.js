import React from 'react';
import ReactDOM from 'react-dom';
import  Messages from './Messages'
import renderer from 'react-test-renderer'

it('Renders without crashing', ()=>{
const div = document.createElement('div');


ReactDOM.render(<Messages/>, div )
ReactDOM.unmountComponentAtNode(div)
});


describe('Message Component',()=>{
it('Renders the UI as expected', ()=>{
  const tree = renderer
  .create(<Messages name='messages' unread={10}/>)
  .toJSON();
  expect(tree).toMatchSnapshot()

});

it('renders the UI as expected with no unreads', () => {
  const tree = renderer
    .create(<Messages name="Messages" unread={0}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});
});