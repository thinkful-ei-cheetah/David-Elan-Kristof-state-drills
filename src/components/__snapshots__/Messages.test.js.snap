// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component Renders the UI as expected 1`] = `
<div
  className="messages"
>
  messages
  <div
    className="unread_count badge"
  >
    10
  </div>
</div>
`;

exports[`Message Component renders the UI as expected with no unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`Renders the UI as expected 1`] = `
<div
  className="messages"
>
  messages
  <div
    className="unread_count badge"
  >
    10
  </div>
</div>
`;

exports[`renders the UI as expected with no unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
